spec/system/follow_category_spec.rb

```diff
@@ -6,36 +6,36 @@
 
   let!(:theme) { upload_theme_component }
 
-  before do
-    sign_in(user)
-    CategoryUser.set_notification_level_for_category(
-      user,
-      NotificationLevels.all[:regular],
-      category.id,
-    )
-  end
+  before { sign_in(user) }
 
   context "when visiting the categories page" do
-    before { visit "/categories" }
+    it "mutes the category when clicking the toggle" do
+      CategoryUser.set_notification_level_for_category(
+        user,
+        NotificationLevels.all[:regular],
+        category.id,
+      )
 
-    it "displays the category toggle button" do
-      expect(page).to have_css("[data-category-id='#{category.id}'] .custom-category-toggle")
-    end
+      visit "/categories"
 
-    it "mutes the category when clicking the toggle" do
       find("[data-category-id='#{category.id}'] .custom-category-toggle").click
 
-      try_until_success do
-        expect(CategoryUser.lookup(user, :muted).pluck(:category_id)).to include(category.id)
-      end
+      expect(CategoryUser.lookup(user, :muted).pluck(:category_id)).to include(category.id)
     end
 
     it "unmutes the category when clicking the toggle again" do
```

Comment:
This test was incorrectly written because we never set the category to muted in the first place. Therefore, clicking on the toggle button doesn't actually mutes it instead of unmuting it.